Dataset support

aviadlazar opened the pull request:
Stable datasets. The code will concat datasets of different videos for ease of use, while still allowing to iterate on each video separately.

sharon200102 committed:
initial version of new datasets

sharon200102 committed:
first version of the new datasets, wasn't tested for correctness

aviadlazar committed:
1. Added pybboxes to setup.py script
2. The test script now loads the collection dataset and iterates over
the frames and videos
3. base.py now uses convert_bboxes for format conversion
4. updated datasets.py, now it concats all the videos datasets for
easier use. It is still possible to iterate on each video seperately.

sharon200102 commented on SoiUtils/datasets/datasets.py:
```diff
@@ -0,0 +1,66 @@
+    
+
+class ImageDetectionDatasetCollection(Dataset):
+    def __init__(self, collection_root_dir: str, annotation_file_name: str, **kwargs) -> None:
```
Why would you assume that the annotation file name is identical in each of the datasets, I think annotation_file_name should be renamed and accept a list of names.

sharon200102 replied:
Great work!

aviadlazar replied:
Solved in the next commit

aviadlazar committed:
When creating a collection of datasets, each dataset can have
a different name for its annotation file.

aviadlazar committed:
The dataset can now be exported as a COCO dataset which
can be used by external repositories that require
a known format.

Added fiftyone to setup.py

It is possible to export to any type that fiftyone supports
but that might be an unnecessary feature

The ability to create a dataset with subsets (train, validation, etc.)
is not tested yet.

aviadlazar committed:
Added fiftyone to setup.py

aviadlazar committed:
The export functionality is now implemented outside of the
CollectionDataset object, so it would be possible to pass
multiple collection datasets that corresponds to different
data subsets.

Subsets are defined through the tags and can be anything.

sharon200102 commented on SoiUtils/datasets/datasets.py:
```diff
@@ -0,0 +1,165 @@
+    def num_subsets(self):
+        return len(self.collection_items_root_dirs)
+
+train_dataset = ImageDetectionDatasetCollection(Path('test_dataset/train'), ["annotations.json"]*4)
```
Please remove these lines of code, I think they were meant for testing.

sharon200102 commented on SoiUtils/test.py:
This test file is not necessary any more, I think it should be removed.

aviadlazar committed:
Delete SoiUtils/test.py

aviadlazar committed:
Removed testing code

aviadlazar committed:
Merge branch 'dataset_support' of https://github.com/sharon200102/soi_utils into dataset_support
